File: flutter/lib/data/post/postService.dart
import 'dart:convert';
import '../../domain/post/reaction.dart';
import '../../domain/post/post.dart';
import '../httpClient.dart';

class PostService {
  Future<Post> createPost(Post post) async {
    HttpClient client = HttpClient();
    Map<String, dynamic> response = await client.post("/post", json.encode(post.toJson()));
    return Post.fromJson(response);
  }

  Future<List<Post>> getPosts() async {
    HttpClient client = HttpClient();
    List<dynamic> response = await client.get("/post");
    return response.map((model) {
      return Post.fromJson(model);
    }).toList();
  }

  Future<Post> reactToPost(Reaction reaction) async {
    HttpClient client = HttpClient();
    String postId = reaction.post.id;
    Map<String, dynamic> response = await client.post("/post/$postId/reaction", json.encode(reaction.toJson()));
    return Post.fromJson(response);
  }
}

File: react-native/data/PostService.ts
import Post from "../domain/post/Post";
import Axios from "axios";
import Reaction from "../domain/post/Reaction";
import Service from "./Service";

export default class PostService extends Service {
    async getAll(): Promise<[Post]> {
        const response = await Axios.get(`${this.HOST}/post`);
        return response.data.map((post: any) => {
            return Post.fromJson(post);
        });
    }

    async createPost(post: Post): Promise<Post> {
        const result = await Axios.post(`${this.HOST}/post`, post);
        return Post.fromJson(result.data);
    }

    async reactToPost(reaction: Reaction): Promise<Post> {
        const url = `${this.HOST}/post/${reaction.postId}/reaction`;
        const result = await Axios.post(url, reaction);
        return Post.fromJson(result.data);
    }
}